src/Services/PatientTransactionService.php

```diff
@@ -0,0 +1,292 @@
+<?php
+
+/**
+ * PatientTransaction Service
+ *
+ * @package   OpenEMR
+ * @link      http://www.open-emr.org
+ * @author    Jonathan Moore <Jdcmoore@aol.com>
+ * @copyright Copyright (c) 2022 Jonathan Moore <Jdcmoore@aol.com>
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+namespace OpenEMR\Services;
+
+//require_once("$srcdir/transactions.inc");
+
+use MongoDB\Driver\Query;
+use OpenEMR\Common\Database\QueryUtils;
+use OpenEMR\Common\Uuid\UuidRegistry;
+use OpenEMR\Services\Search\DateSearchField;
+use OpenEMR\Services\Search\TokenSearchField;
+use OpenEMR\Services\Search\TokenSearchValue;
+use OpenEMR\Validators\ProcessingResult;
+use Particle\Validator\Exception\InvalidValueException;
+use Particle\Validator\Validator;
+use OpenEMR\Validators\BaseValidator;
+
+class PatientTransactionService extends BaseService
+{
+    const TABLE_NAME = "transactions";
+
+    public function __construct()
+    {
+        parent::__construct(self::TABLE_NAME);
+    }
+
+    public function getSelectStatement($predicateColumnName){
+        return 
+        "
+        SELECT
+            d.form_id,
+            t.title,
+            t.groupname,
+            t.authorized,
+            MAX(
+                CASE WHEN d.field_id = 'body' THEN d.field_value ELSE ''
```

Comment:
Rather than hard-coding every attribute of the LBT, have your tried building a JSON object?

```
SELECT form_id, GROUP_CONCAT(CONCAT(QUOTE(field_id),":",QUOTE(field_value)) SEPARATOR ",") lbt_values
FROM lbt_data
GROUP BY form_id
```